docs/operate/invocation.mdx

```diff
@@ -9,286 +9,309 @@ import Admonition from '@theme/Admonition';
-grpcurl -plaintext <RESTATE_INGRESS_URL> describe
-```
 
-Restate also natively supports gRPC-web. You can use a [gRPC-web code generator](https://www.npmjs.com/package/grpc-web) and point it directly to Restate, without using a 3rd party proxy to translate gRPC-web to gRPC.
 
-### Invoke a service idempotently
+### Invoke a handler idempotently
 
 You can send requests to Restate providing an idempotency key, through the [`Idempotency-Key` header](https://datatracker.ietf.org/doc/draft-ietf-httpapi-idempotency-key-header/):
 
 ```shell
-curl <RESTATE_INGRESS_URL>/org.example.Greeter/Greet \
-  -H 'idempotency-key: <IDEMPOTENCY_KEY>' \
+curl localhost:8080/MyService/myHandler \
+# focus
+  -H 'idempotency-key: ad5472esg4dsg525dssdfa5loi' \
   -H 'content-type: application/json' \
-  -d '{"name": "Restate"}'
+  -d '{"name": "Mary", "age": 25}'
 ```
 
-After the invocation completes, Restate persists the response for a retention period of 30 minutes.
-If you re-invoke the service with the same idempotency key within 30 minutes, Restate sends back the same response and doesn't re-execute the request to the service.
-
-You can tune the retention period by setting the header `idempotency-retention-period: <seconds>`
+After the invocation completes, Restate persists the response for a retention period of one day (24 hours).
+If you re-invoke the service with the same idempotency key within 24 hours, Restate sends back the same response and doesn't re-execute the request to the service.
 
-The `Idempotency-Key` header works with both HTTP and gRPC/gRPC-web.
-
-### Invoke a service without waiting for the response
-
-You can invoke a service without waiting for the response, similar to one-way calls in the SDK ([TypeScript docs](/develop/ts/service-communication#one-way-calls) or [Java docs](/develop/java/service-communication#sending-messages)), by using the Restate built-in `dev.restate.Ingress/Invoke` service method, which can be invoked like any other user service, using gRPC, gRPC-web or Connect.
+<Admonition type="tip" title={"Make any service call idempotent by using Restate"}>
+    By using Restate and an idempotency key, you can make any service call idempotent, without any extra code or setup.
+    This is a very powerful feature to ensure that your system stays consistent and doesn't perform the same operation multiple times.
+</Admonition>
 
-For example, using HTTP and `curl`:
+<details className={"grey-details"}>
+    <summary> Tuning retention time</summary>
+
+    <CH.Section>
+    You can tune the [retention time](focus://3) on a service-level by using the [Admin API](focus://1[15:28]) ([docs](/references/admin-api#tag/service/operation/modify_service)):
+    ```shell
+    curl -X PATCH localhost:9070/services/MyService \
+        -H 'content-type: application/json' \
+        -d '{"idempotency_retention": "2days"}'
+    ```
+    The [retention time](focus://3) is in [humantime format](https://docs.rs/humantime/latest/humantime/).
```

Comment:
It seems that we have at least 3 ways to specify a duration: 1. ISO8601, 2. secs 3. humantime format. I am wondering whether there is value in unifying those. What do you think @slinkydeveloper?

Reply:
I'm aware of this, i'll address it at some point. I wanna stay flexible and support all three formats, e.g. try ISO8601 first, then try humantime, then fallback to try to parse as number.

Reply:
Should we start with a single one and then try to see what users ask for? This might simplify configuration and operation for users because they don't have to remember what format is expected where.

Reply:
> Should we start with a single one and then try to see what users ask for?

There's very good reasons to have all 3 of them, in particular there's good reason to support ISO8601 at ingress, and humantime at admin api. ISO8601 is supported in most languages OOTB, and generally ingress requests will be programmatic, and not manual, so we should support a format that is widely adopted. Admin API on the other hand might be invoked by humans more likely, so using humantime there makes sense.

I'll change this after 0.9, changing it now requires me to change the SDKs typed clients too.